lib/dor/text_extraction/speech_to_text.rb

```diff
@@ -52,8 +52,42 @@ def cleanup_s3_folder
       # return a list of filenames that should be stt'd
       # iterate over all files in cocina_object.structural.contains, looking at mimetypes
       # return a list of filenames that are correct mimetype
+      # then filter out any files that either (1) do not have an audio track or (2) have audio that is mostly silent
       def filenames_to_stt
-        stt_files.map(&:filename)
+        available_files = stt_files.map(&:filename)
+        available_files.select { |filename| has_audio_track?(filename) && has_useful_audio_track?(filename) }
+      end
+
+      # determines if the given file has an audio track by parsing the technical metadata and looking at the file's audio_count
+      def has_audio_track?(filename)
+        file_level_tech_metadata(filename)&.dig('av_metadata', 'audio_count')&.positive? || false
+      end
+
+      # using new technical metadata generated in https://github.com/sul-dlss/technical-metadata-service/pull/572
+      # this uses the audio max_volume and mean_volume fields to determine if the audio is mostly silent
+      def has_useful_audio_track?(filename)
+        audio_metadata = file_level_tech_metadata(filename)&.dig('dro_file_parts')&.find { |parts| parts['part_type'] == 'audio' }&.dig('audio_metadata')
+        return false unless audio_metadata && audio_metadata['max_volume'] && audio_metadata['mean_volume']
+
+        audio_metadata['mean_volume'] > -40 && audio_metadata['max_volume'] > -30
+      end
+
+      # return the technical metadata for a given filename
+      def file_level_tech_metadata(filename)
+        tech_metadata.find { |file| file['filename'] == filename }
+      end
+
+      # return the technical metadata for the object from the technical-metadata-service and parse it as json
+      def tech_metadata
+        @tech_metadata ||= begin
+          resp = Faraday.get("#{Settings.tech_md_service.url}/v1/technical-metadata/druid/#{cocina_object.externalIdentifier}") do |req|
+            req.headers['Content-Type'] = 'application/json'
+            req.headers['Authorization'] = "Bearer #{Settings.tech_md_service.token}"
+          end
+          raise "Technical-metadata-service returned #{resp.status} when requesting techmd for #{bare_druid}: #{resp.body}" unless resp.success?
+
+          JSON.parse(resp.body)
+        end
```

Comment:
this is the code that does that detection by examining technical metadata

note that volume detection level technical metadata is dependent on https://github.com/sul-dlss/technical-metadata-service/pull/572 being merged first and technical metadata updated on any object that needs it